CMakeLists.txt

```diff
@@ -15,7 +15,7 @@
 cmake_minimum_required(VERSION 3.13)
 project(stim)
 include_directories(src)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 17)
```

Comment:
This is a big change that should probably be done separately. You can see I have a C++20 variant that's just been sitting... because I'm terrified of merging it and breaking anything about the python packaging flow.